File: 202-happy-number/happy-number.kt
class Solution {
    fun isHappy(n: Int): Boolean {
        val seenNumbers = mutableSetOf<Int>()
        var current = n

        while (current != 1) {
            if (seenNumbers.contains(current)) return false
            seenNumbers.add(current)
            current = getSumOfSquares(current)
        }

        return true
    }

    private fun getSumOfSquares(num: Int): Int {
        var sum = 0
        var number = num
        while (number != 0) {
            val digit = number % 10
            sum += digit * digit
            number /= 10
        }
        return sum
    }
}

File: 202-happy-number/happy-number.py
class Solution(object):
    def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 1:
            return True
        seen = set()
        while True:
            if n in seen:
                return False
            seen.add(n)
            temp = 0
            while n:
                temp += (n%10)**2
                n //= 10
            if temp == 1:
                return True
            n = temp
